### DISCUSSION TITLE
Feature Request: from_json (or from_json_auto?) should default to the structure returned by json_structure

### DISCUSSION THREAD

**Discussion by gotdan** [duckdb/duckdb#8215] [Ideas]
It seems like `from_json` (or a `from_json_auto` variant) should default to structure based on `json_structure` so that the result of `from_json('{"a": 1}')` would be equivalent to that of `from_json('{"a": 1}', json_structure('{"a": 1}'))`. It would also be nice if casting a json type to a struct type worked this way if no schema is passed in. Does this makes sense, or is there already a good way to do this in duckdb? Thanks!

**Comment by lnkuiper** [comment]
Hi @gotdan, thanks for the suggestion. I have thought about this quite a bit, but haven't found a way yet to implement this properly (yet).

The issue here is that the column referenced in `from_json` could potentially be from a table with thousands of rows, in which case we'd need to sample the JSON schema from all of these rows to infer the JSON structure, before executing the `from_json`.

One thing that now comes to mind is that we could potentially keep statistics on columns of type `JSON`, which would allow us to just get the structure from the statistics. This would allow `from_json` to be performed without supplying a structure, but only for DuckDB tables.

Anyways, that's just my thoughts for now, I will think about it some more.

**Comment by gotdan** [comment]
Hi @lnkuiper, thanks for the reply! 

Yup, that makes sense around the performance challenges of this approach in larger datasets. 

Your idea of storing stats on the structures in a json column is really neat though! If there was then a way to then output this schema info in a pragma query, folks could also use DuckDB as a tool to easily characterize a set of json data. This seems broadly useful - for example, one could use the stats to identify unexpected properties and types in a dataset, or even auto generate json-schema validation profiles based on the properties of real world data.

**Comment by bayre** [comment]
I agree that this would be a great feature! Noted support for it in [this issue](https://github.com/duckdb/duckdb/issues/7765#issuecomment-1573700523) too. For the time being I've resorted to running my query twice — first for structure and then for transformation.
